hotshot-query-service/src/availability.rs

```diff
@@ -161,9 +166,97 @@ impl Error {
     }
 }
 
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(bound = "")]
+pub struct Leaf1QueryData<Types: NodeType> {
+    pub(crate) leaf: Leaf<Types>,
+    pub(crate) qc: QuorumCertificate<Types>,
+}
+
+fn downgrade_leaf<Types: NodeType>(leaf2: Leaf2<Types>) -> Leaf<Types> {
+    // TODO do we still need some check here?
+    // `drb_seed` no longer exists on `Leaf2`
+    // if leaf2.drb_seed != [0; 32] && leaf2.drb_result != [0; 32] {
+    //     panic!("Downgrade of Leaf2 to Leaf will lose DRB information!");
+    // }
```

Comment:
@ss-es @lukaszrzasik maybe you can comment on this? I think the downgrade is mainly used in the query service now when paths to API v0 are invoked?

On a related note, I think we should **not** implement `From` when the conversion is lossy. This is the second bullet point here: https://doc.rust-lang.org/std/convert/trait.From.html#when-to-implement-from

I think if the conversion is lossy it's better to have a function or method, this then also makes it easier to search the codebase for these lossy conversions.

https://github.com/EspressoSystems/espresso-sequencer/blob/e15b2b8f6a2d8eb542505f36809cfcc07945a1b1/hotshot-query-service/src/availability.rs#L277-L301

cc @imabdulbasit